jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -456,19 +456,13 @@ List<Image> getCachedBaseImages()
 
     if (manifestList == null) {
       Verify.verify(manifestsAndConfigs.size() == 1);
-      ManifestTemplate manifest = manifestsAndConfigs.get(0).getManifest();
-      if (manifest instanceof V21ManifestTemplate) {
-        return Collections.singletonList(
-            JsonToImageTranslator.toImage((V21ManifestTemplate) manifest));
+      ManifestAndConfigTemplate manifestAndConfig = manifestsAndConfigs.get(0);
+      Optional<Image> cachedImage = getBaseImageIfAllLayersCached(manifestAndConfig, true);
+      if (!cachedImage.isPresent()) {
+        return Collections.emptyList();
+      } else {
+        return Collections.singletonList(cachedImage.get());
       }
```

Comment:
super nit: we can do away with the `else` here and return `Collections.singletonList(...)` outside the `if`.